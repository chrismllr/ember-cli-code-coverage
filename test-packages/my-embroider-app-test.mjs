'use strict';

import { execa } from 'execa';
import setupTestDir, { assertCoverageExists, assertFileIsNotEmpty } from './utils.mjs';
import { describe, it } from 'vitest';

const APP_DIR  = 'my-embroider-app';

describe('app coverage generation', function () {
  const env = { COVERAGE: 'true' };

  for (const embroiderVersion of [
      '^0.47.0',
      '^1.0.0',
      '^2.1.0', 
      '^3.1.0'
    ]) {
    it(`generates coverage with @embroider ${embroiderVersion}`, async function (context) {
      let buildPath = await setupTestDir(APP_DIR, env, {
        "@embroider/compat": embroiderVersion,
        "@embroider/core": embroiderVersion,
        "@embroider/webpack": embroiderVersion
      });
  
      await execa('npx',  ['ember','test', '--test-port=0'], { cwd: buildPath, env });
  
      await assertCoverageExists(`${buildPath}/coverage`);
    });
  }
});
