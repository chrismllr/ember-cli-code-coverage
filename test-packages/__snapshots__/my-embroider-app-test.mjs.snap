// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`app coverage generation > generates coverage with @embroider ^0.47.0 1`] = `
{
  "app/app.js": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "lines": {
      "covered": 4,
      "pct": 100,
      "skipped": 0,
      "total": 4,
    },
    "statements": {
      "covered": 4,
      "pct": 100,
      "skipped": 0,
      "total": 4,
    },
  },
  "app/router.js": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "lines": {
      "covered": 1,
      "pct": 33.33,
      "skipped": 0,
      "total": 3,
    },
    "statements": {
      "covered": 1,
      "pct": 33.33,
      "skipped": 0,
      "total": 3,
    },
  },
  "total": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "branchesTrue": {
      "covered": 0,
      "pct": "Unknown",
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "lines": {
      "covered": 5,
      "pct": 71.42,
      "skipped": 0,
      "total": 7,
    },
    "statements": {
      "covered": 5,
      "pct": 71.42,
      "skipped": 0,
      "total": 7,
    },
  },
}
`;

exports[`app coverage generation > generates coverage with @embroider ^1.0.0 1`] = `
{
  "app/app.js": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "lines": {
      "covered": 4,
      "pct": 100,
      "skipped": 0,
      "total": 4,
    },
    "statements": {
      "covered": 4,
      "pct": 100,
      "skipped": 0,
      "total": 4,
    },
  },
  "app/router.js": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "lines": {
      "covered": 1,
      "pct": 33.33,
      "skipped": 0,
      "total": 3,
    },
    "statements": {
      "covered": 1,
      "pct": 33.33,
      "skipped": 0,
      "total": 3,
    },
  },
  "total": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "branchesTrue": {
      "covered": 0,
      "pct": "Unknown",
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "lines": {
      "covered": 5,
      "pct": 71.42,
      "skipped": 0,
      "total": 7,
    },
    "statements": {
      "covered": 5,
      "pct": 71.42,
      "skipped": 0,
      "total": 7,
    },
  },
}
`;

exports[`app coverage generation > generates coverage with @embroider ^2.1.0 1`] = `
{
  "app/app.js": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "lines": {
      "covered": 4,
      "pct": 100,
      "skipped": 0,
      "total": 4,
    },
    "statements": {
      "covered": 4,
      "pct": 100,
      "skipped": 0,
      "total": 4,
    },
  },
  "app/router.js": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "lines": {
      "covered": 1,
      "pct": 33.33,
      "skipped": 0,
      "total": 3,
    },
    "statements": {
      "covered": 1,
      "pct": 33.33,
      "skipped": 0,
      "total": 3,
    },
  },
  "total": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "branchesTrue": {
      "covered": 0,
      "pct": "Unknown",
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "lines": {
      "covered": 5,
      "pct": 71.42,
      "skipped": 0,
      "total": 7,
    },
    "statements": {
      "covered": 5,
      "pct": 71.42,
      "skipped": 0,
      "total": 7,
    },
  },
}
`;
